app.rb

```diff
@@ -4,128 +4,40 @@
 require './book'
 require './classroom'
 require './rental'
+require './lib/create_books'
+require './lib/create_persons'
+require './lib/create_rental'
+require './lib/displayer'
 
 class App
   def initialize
     @books = []
     @persons = []
     @rentals = []
-  end
-
-  def start_console
-    puts 'Welcome to my School Library!'
-    until list_of_options
-      input = gets.chomp
-      if input == '7'
-        puts 'Thank you for using my School Library!'
-        break
-      end
-
-      option input
-    end
+    @displayer = Displayer.new
   end
 
   def list_all_books
-    puts 'Database is empty! Add a book.' if @books.empty?
-    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+    @displayer.display_books(@books)
   end
 
   def list_all_persons
-    puts 'Database is empty! Add a person.' if @persons.empty?
-    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name} Age: #{person.age} Id: #{person.id}" }
+    @displayer.display_persons(@persons)
   end
 
   def create_person
-    print 'To create a student, press 1, to create a teacher, press 2 :'
-    option = gets.chomp
-
-    case option
-    when '1'
-      create_student
-    when '2'
-      create_teacher
-    else
-      puts 'Invalid input. Try again'
-    end
-  end
-
-  def create_student
-    puts 'Create a new student'
-    print 'Enter name: '
-    name = gets.chomp
-    print 'Enter student age: '
-    age = gets.chomp.to_i
-    print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp.downcase
-    case parent_permission
-    when 'n'
```

Comment:
- [ ] Please let's make our codes simple (KISS). All requests for inputs and input validations should have been placed somewhere else or in main.rb file. Also, remember about the best practice and SOLID lessons.

Reply:
These are the deleted code